static/js/silhouette.js

```diff
@@ -0,0 +1,142 @@
+'use strict'
+
+/**
+ * Calculate euclidean distance between two points
+ *
+ * @function euclideanDist
+ * @param {array} x - coordinate of point x
+ * @param {array} y - coordinate of point y
+ * @return {number} euclidean distance between x and y
+ */
+function euclideanDist(x, y) {
+  // check x, y
+  if (x == y) {
+    return 0;
+  }
+  let sum = 0;
+  for (let i = 0; i < x.length; i++) {
+    sum += (x[i] - y[i]) ** 2;
+  }
+  return Math.sqrt(sum);
+}
+
+/**
+ * return the pairwise distance matrix of each point in the input data array
+ *
+ * @function pairwiseDist
+ * @param {Array} x - the input data array
+ * @return {Array} distance matrix of the given data
+ */
+//TODO: add metrics='euclideanDist'
+function pairwiseDist(x) {
+  let d = Array(x.length).fill().map(()=>Array(x.length).fill());
+  for (let i = 0; i < x.length; i++) {
+    for (let j = 0; j < x.length; j++) {
+      if (j === i) {
+        d[i][j] = 0;
+      } else if (j < i) {
+        d[i][j] = d[j][i];
+      } else {
+        d[i][j] = euclideanDist(x[i], x[j]);
+      }
+    }
+  }
+  return d;
+}
+
+
+/**
+ * return the occurrence of each entry in the input data
+ *
+ * @function bincount
+ * @param {Array} x - the input data array
+ * @return {Array} the occurrence of each entry in the input data
+ */
+function bincount(x) {
+  let res = Array(Math.max.apply(Math, x) + 1).fill(0);
+  for (let i = 0; i < x.length; i++) {
+    res[x[i]]++;
+  }
+  return res;
+}
+
+
+/**
+ * calculate the average of each element in the input data array
+ *
+ * @function mean
+ * @param {Array} x - the input data array
+ * @return {Number} the average value of each element of the array
```

Comment:
According to [JSDoc](https://jsdoc.app/tags-param.html), primitive types (e.g., `string` and `number`) should be all lowercase, `Array` and `Object` should be capitalized. Optionally, one can specify the type of array elements: `number[]` instead of just `Array`.